code/game/machinery/vending/cm_vending.dm

```diff
@@ -952,6 +952,18 @@ IN_USE						used for vending/denying
 	for(R in (stock_listed_products))
 		if(item_to_stock.type == R[3] && !istype(item_to_stock,/obj/item/storage))
 
+			if(istype(item_to_stock, /obj/item/device/defibrillator))
+				var/obj/item/device/defibrillator/D = item_to_stock
+				if(D.dcell.charge < D.dcell.maxcharge)
```

Comment:
future-proof this check, add a "does this defib even have a cell", because i bet someone in three years is gonna add a defib with swappable cells